### ISSUE TITLE
Switch from special type names (sexp_{list,bool,array,option}) to attributes

### ISSUE THREAD

**Issue by copy** [janestreet/core_kernel#80]
Opening Core_kernel or Base bring the type aliases  `sexp_{list,bool,array,option}` to `{list,bool,array,option}` into scope, which infamously infect OCaml tooling (merlin, utop) and confuse newcomers. See also https://github.com/janestreet/base/issues/39.

`ppx_sexp_conv` already supports [attributes](https://github.com/janestreet/ppx_sexp_conv#defaults) that supersede the old type names. So we should encourage users to switch to attributes and fix the original problem by moving the type aliases out of `Core_kernel` and `Base` (maybe to a submodule?).

**Comment by diml** [comment]
Indeed, we have been talking about doing this change for a while. It's just that we never took the time to do it.

Regarding moving the types aliases to a sub-module, for the user this is basically as much work as switching to attributes. We could mark them deprecated with `[@@deprecated]` though.

**Comment by copy** [comment]
It would be wonderful if 2019 could be the year in which we don't have to see `sexp_list` any more. The more one uses merlin, the more often they come up, which is quite unfortunate.

**Comment by yminsky** [comment]
Yeah, it's pretty annoying. It's on our near-term todo list.

**Comment by kevinji** [comment]
See b415c54204224e09fec4763ff4f3398fdfa943a2!

**Comment by seliopou** [comment]
These have been deprecated for a while in base and core_kernel, so I'm gonna go ahead and delete them entirely.

**Comment by copy** [comment]
This has been fixed a while ago, thanks!
